src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -37,9 +39,11 @@
                    (request-handler request))
 
     (comidi/GET ["/catalog/" [#".*" :rest]] request
-                   (request-handler request))
+                   ((request-core/catalog-request-handler
+                     request-handler get-code-id-fn) request))
     (comidi/POST ["/catalog/" [#".*" :rest]] request
-                    (request-handler request))
+                    ((request-core/catalog-request-handler
+                      request-handler get-code-id-fn) request))
```

Comment:
This is a bit hard to parse, what with the invocation of a function that returns a function which is then invoked.  Perhaps something like that
```clj
(let [request-handler-with-code-id (request-core/catalog-request-handler request-handler get-code-id-fn)]
  (request-handler-with-code-id request)
```
... or something like that.